cowprotocol/services
# Internal buffer trading quotes

# Description
See linked #2345 for more context about this change.

# Changes
Added the needed machinery to let external solvers internalize interactions for quotes.

Moved trusted tokens arguments into the shared arguments so `autopilot` and `orderbook` both have access to them.
Updated code that builds the `AutoUpdatingTokenList` and passes it around.

Slightly adjusted the `local_node_buffers` e2e test to see that we can compute quotes based on internal buffer trades.

## How to test
Updated existing e2e test to make sure the new format doesn't cause issues.
In a future PR we could try to add internalization in the baseline logic and verify that the existing e2e test for internalization behaves accordingly.
